File: backend/src/main/java/com/utsman/backend/repository/DriverRepository.java
package com.utsman.backend.repository;

import com.utsman.backend.model.Driver;

import org.bson.types.ObjectId;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface DriverRepository extends MongoRepository<Driver, String> {
    Driver findByEmail(String email);
    Driver findBy_id(ObjectId _id);
}

File: backend-side/src/main/kotlin/com/kemana/backend/repository/DriverRepository.kt
package com.kemana.backend.repository

import com.kemana.backend.model.Driver
import org.springframework.data.mongodb.repository.MongoRepository

interface DriverRepository : MongoRepository<Driver, String> {
    fun findDriverById(id: String) : Driver?
    fun findDriverByEmail(email: String) : Driver
}